ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/DescribeConnectorExecutor.java

```diff
@@ -106,13 +113,39 @@ public DescribeConnectorExecutor() {
       sources = ImmutableList.of();
     }
 
+    List<KsqlWarning> warnings;
+    List<String> topics;
+    try {
+      topics = getTopics(serviceContext.getAdminClient(), connector);
+      warnings = ImmutableList.of();
+    } catch (Exception e) {
+      topics = ImmutableList.of();
+      warnings = ImmutableList.of(
+          new KsqlWarning("Could not list related topics due to " + e.getMessage())
+      );
+    }
+
     final ConnectorDescription description = new ConnectorDescription(
         configuredStatement.getStatementText(),
         info.config().get(ConnectorConfig.CONNECTOR_CLASS_CONFIG),
         status,
-        sources
+        sources,
+        topics,
+        warnings
```

Comment:
Looks like `ConnectorDescription` is only ever created with _either_ the topics list _or_ a single warning.

Might be worth codifying this. i.e. the `topics` param passed in to `ConnectorDescription` should be a union type of _either_ the topic list _or_ a single warning.

Reply:
In the future we might want to supply other warnings that are unrelated to the list of topics (e.g. failed to connect to Connect) - I don't think that coupling necessarily makes sense.